### PR TITLE
New: Tag component

### PR DESCRIPTION
**Pull request by omgaz**
- Tags can have accent colours and be extended with custom colours
- Tags can be inverse
- Tags can be inline
- Tags ellipse after about 18 characters
- Increased max-statements allowed to 10 (eslint recommended default)*

---
screen shot 2016-07-26 at 09 51 49
---

\* `max-statements` was causing errors, the component is accepting too many properties. Although it's nice to keep them simple and unmodifiable, I feel there are exceptions to the rules. Let me know if this component is too complex and if we would prefer to have **two** components; a \<ClearableTag\> and a \<Tag\> component.

### CODE REVIEW

**Comment by omgaz** [comment]
Linting is failing because the tag component exceeds the max statments level of 6. Will look into this tomororw.

**Comment by vinteo** [comment]
LGTM

**Comment by pphminions** [comment]
LGTM 🍒

File: src/components/alexandria/TagComponent.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+TagComponent.propTypes = {
+  children: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  accent: PropTypes.string,
+  inverse: PropTypes.bool,
+  clearable: PropTypes.bool,
+  onClear: PropTypes.func,
+  clearIconSvgHref: PropTypes.string,
+};
```
Comment (NamrataTrivedi) [inline src/components/alexandria/TagComponent.js]: will be good to have the props in alphabetical order :)

**Comment by NamrataTrivedi** [comment]
LGTM :candy:

File: src/components/alexandria/TagComponent.js (hunk 1 above)
Comment (omgaz) [inline src/components/alexandria/TagComponent.js]: I've tried to group them as required first. Followed by id (important but not required) then configurable properties of accent, inverse and clearable. Lastly, grouped with the clearable the clear customisations.

**Comment by mcslash** [comment]
Awesome gazman! LGTM 🐪

File: src/components/alexandria/TagComponent.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export const ActionButton = ({ onAction, id, actionIconSvgHref }) => (
+  <span className="action-button" onClick={() => onAction(id)}>
+    <SvgSymbol href={actionIconSvgHref} />
+  </span>
+);
+
+ActionButton.propTypes = {
+  id: PropTypes.string,
+  onAction: PropTypes.func,
```
Comment (patrick-ivers) [inline src/components/alexandria/TagComponent.js]: Shouldn't this be required? It's not really an "action" button if it doesn't do anything.

File: src/components/alexandria/TagComponent.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+TagComponent.displayName = 'AlexandriaTagComponent';
+
+TagComponent.propTypes = {
+  children: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  accent: PropTypes.string,
+  inverse: PropTypes.bool,
+  actionable: PropTypes.bool,
+  onAction: PropTypes.func,
```
Comment (patrick-ivers) [inline src/components/alexandria/TagComponent.js]: Could the presence of an onAction function indicate that a Tag is "actionable"?

File: test/components/alexandria/TagComponentTest.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import _ from 'lodash';
+import Tag, { ActionButton } from 'components/alexandria/TagComponent';
+import SvgSymbol from 'components/alexandria/SvgSymbolComponent';
+import React from 'react';
+import sinon from 'sinon';
+
+describe('TagComponent', () => {
+  it('should render a default tag', () => {
+    const component = shallow(<Tag>You are it!</Tag>);
+    expect(component.find('span').prop('className')).to.eql('tag-component');
```
Comment (patrick-ivers) [inline test/components/alexandria/TagComponentTest.js]: Is it better to use `.equal`, which is strict equality (===), over `.eql`, which isn't?

**Comment by mcslash** [comment]
Merging!

**Comment by mcslash** [comment]
:shipit:

**Comment by patrick-ivers** [comment]
Oi! Beat me to it! 👊

**Comment by patrick-ivers** [comment]
FWIW, LGTM! 🍰

**Pull request merged**